integration/tests/cook/test_basic.py

```diff
@@ -1538,40 +1497,6 @@ def test_error_while_creating_job(self):
         self.assertEqual(resp.status_code, 400)
         self.assertEqual(f'Duplicate job uuids: ["{job1["uuid"]}"]', resp.json()['error'], resp.text)
 
-    @unittest.skipIf(util.has_ephemeral_hosts(), util.EPHEMERAL_HOSTS_SKIP_REASON)
-    @pytest.mark.xfail
-    def test_hostname_equals_job_constraint(self):
```

Comment:
Relies on `util.hostnames_to_consider`, is specific to non-autoscaling environments, and is already marked `xfail`